bom/runtime/pom.xml

```diff
@@ -904,6 +904,11 @@
                         <groupId>io.netty</groupId>
                         <artifactId>netty-all</artifactId>
                     </exclusion>
+                    <exclusion>
+                        <!-- explicitly exclude the vert.x dependency to use the version defined by Quarkus -->
+                        <groupId>io.vertx</groupId>
+                        <artifactId>vertx-core</artifactId>
```

Comment:
What I was saying is that this shouldn't be necessary as we include the Vert.x bom at the beginning of this bom.

If it is an issue, then there's something going wrong.

Reply:
Your BOM won't change the version imported by a dependency.

Reply:
I may have a solution, but that may end up ugly.

Reply:
Actually nope.

What could be done is to have an _empty_ module in quarkus that depends on Vert.x core and quarkus-http (in this order), and all other modules would depend on this module.

As there is no easy solution, we probably should remove the `exclusion` and manage the alignment by hand for now. That would remove the dependencies added to the lambda modules.

Reply:
> Your BOM won't change the version imported by a dependency.

Hu? Are we talking about the same thing? A BOM is part of dependency management so it will apply a consistent version across all your dependencies.

Reply:
Here, quarkus-http was depending on vert.x 3.8.0.
Even if the bom says that the Vert.x version is 3.8.1, if a module depends on quarkus-http, and does not explicitly depends on vert.x, it will resolve to 3.8.0.

Reply:
That was actually the first build issue. Even with the BOM, it was resolving to Vert.x 3.8.0.

Reply:
I've removed the exclusion.

Reply:
Then we have an issue with the BOM because it should enforce consistently the Vert.x version across all the codebase, including in your case. That's what dependency management is for. So if it doesn't work, we have something to fix with the Vert.x BOM inclusion.

Reply:
Then we need to merge quarkus-http in the main codebase or implement the _empty_ module trick.

Reply:
No, it should work as is. That's what I'm trying to say. Dependency management will enforce consistency across all your dependencies and the dependencies of your dependencies (and so on).

I trust you that there was an issue, I'll check later what's going on with the BOM.

Reply:
So right now, it enforce the vert.x version in the quarkus codebase which is what we want.